### ISSUE TITLE
Wrong number of arguments: (3 . 4), 2 when starting emacs

### ISSUE THREAD

**Issue by c02y** [syl20bnr/spacemacs#14640]
Since this issue happens before `report-issue` of spacemacs is loaded, so I cannot report this issue using spacemacs itself.

Reproduce steps:
1. clean clone `git clone -b develop https://github.com/syl20bnr/spacemacs ~/.emacs.d`
2. `cp ~/.emacs.d/core/templates/.spacemacs.template ~/.spacemacs`
3. edit ~/.spacemacs, add `c-c++` layer
4. start emacs to install all the pakcages
5. restart emacs

Error message:
```
Warning (initialization): An error occurred while loading ‘/home/chz/.emacs.d/init.el’:

Wrong number of arguments: (3 . 4), 2

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace. Disable showing Disable logging

```
Error message of `emacs --debug-init`:
```
Debugger entered--Lisp error: (wrong-number-of-arguments (3 . 4) 2)
  #f(compiled-function (obsolete-name current-name when &optional docstring) "Make OBSOLETE-NAME a variable alias for CURRENT-NAME and mark it obsolete.\n\nWHEN should be a string indicating when the variable was first\nmade obsolete, for example a date or a release number.\n\nThis macro evaluates all its parameters, and both OBSOLETE-NAME\nand CURRENT-NAME should be symbols, so a typical usage would look like:\n\n  (define-obsolete-variable-alias 'foo-thing 'bar-thing \"27.1\")\n\nThis macro uses `defvaralias' and `make-obsolete-variable' (which see).\nSee the Info node `(elisp)Variable Aliases' for more details.\n\nIf CURRENT-NAME is a defcustom or a defvar (more generally, any variable\nwhere OBSOLETE-NAME may be set, e.g. in an init file, before the\nalias is defined), then the define-obsolete-variable-alias\nstatement should be evaluated before the defcustom, if user\ncustomizations are to be respected.  The simplest way to achieve\nthis is to place the alias statement before the defcustom (this\nis not necessary for aliases that are autoloaded, or in files\ndumped with Emacs).  This is so that any user customizations are\napplied before the defcustom tries to initialize the\nvariable (this is due to the way `defvaralias' works).\n\nFor the benefit of Customize, if OBSOLETE-NAME has\nany of the following properties, they are copied to\nCURRENT-NAME, if it does not already have them:\n`saved-value', `saved-variable-comment'." #<bytecode 0x1e00c00e13bb0281>)('c++-enable-organize-includes-on-save 'c-c++-enable-organize-includes-on-save)
  (define-obsolete-variable-alias 'c++-enable-organize-includes-on-save 'c-c++-enable-organize-includes-on-save)
  eval-buffer(#<buffer  *load*-946121> nil "/home/chz/.emacs.d/layers/+lang/c-c++/config.el" nil t)  ; Reading at buffer position 2544
  load-with-code-conversion("/home/chz/.emacs.d/layers/+lang/c-c++/config.el" "/home/chz/.emacs.d/layers/+lang/c-c++/config.el" nil nil)
  load("/home/chz/.emacs.d/layers/+lang/c-c++/config.el" nil nil)
  configuration-layer/load-file("/home/chz/.emacs.d/layers/+lang/c-c++/config.el")
  (if (file-exists-p file) (configuration-layer/load-file file))
  (let ((file (concat (eieio-oref obj ':dir) file))) (if (file-exists-p file) (configuration-layer/load-file file)))
  (while --dolist-tail-- (setq file (car --dolist-tail--)) (let ((file (concat (eieio-oref obj ':dir) file))) (if (file-exists-p file) (configuration-layer/load-file file))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- files) file) (while --dolist-tail-- (setq file (car --dolist-tail--)) (let ((file (concat (eieio-oref obj ':dir) file))) (if (file-exists-p file) (configuration-layer/load-file file))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (progn (let ((--dolist-tail-- files) file) (while --dolist-tail-- (setq file (car --dolist-tail--)) (let ((file (concat (eieio-oref obj ...) file))) (if (file-exists-p file) (configuration-layer/load-file file))) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  (if obj (progn (let ((--dolist-tail-- files) file) (while --dolist-tail-- (setq file (car --dolist-tail--)) (let ((file (concat ... file))) (if (file-exists-p file) (configuration-layer/load-file file))) (setq --dolist-tail-- (cdr --dolist-tail--))))))
  (let ((obj (configuration-layer/get-layer layer-name))) (if obj (progn (let ((--dolist-tail-- files) file) (while --dolist-tail-- (setq file (car --dolist-tail--)) (let ((file ...)) (if (file-exists-p file) (configuration-layer/load-file file))) (setq --dolist-tail-- (cdr --dolist-tail--)))))))
  configuration-layer//load-layer-files(c-c++ ("config.el"))
  (let ((spacemacs-customization--current-group (spacemacs-customization//create-layer-group layer-name (configuration-layer//get-layer-parent-category layer-name)))) (configuration-layer//load-layer-files layer-name '("config.el")))
  (while --dolist-tail-- (setq layer-name (car --dolist-tail--)) (let ((spacemacs-customization--current-group (spacemacs-customization//create-layer-group layer-name (configuration-layer//get-layer-parent-category layer-name)))) (configuration-layer//load-layer-files layer-name '("config.el"))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- layer-names) layer-name) (while --dolist-tail-- (setq layer-name (car --dolist-tail--)) (let ((spacemacs-customization--current-group (spacemacs-customization//create-layer-group layer-name (configuration-layer//get-layer-parent-category layer-name)))) (configuration-layer//load-layer-files layer-name '("config.el"))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((warning-minimum-level :error)) (let ((--dolist-tail-- layer-names) layer-name) (while --dolist-tail-- (setq layer-name (car --dolist-tail--)) (let ((spacemacs-customization--current-group (spacemacs-customization//create-layer-group layer-name (configuration-layer//get-layer-parent-category layer-name)))) (configuration-layer//load-layer-files layer-name '("config.el"))) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  configuration-layer//configure-layers((spacemacs-bootstrap spacemacs-visual spacemacs-purpose spacemacs-project spacemacs-org spacemacs-navigation spacemacs-modeline spacemacs-misc spacemacs-language spacemacs-evil spacemacs-editing-visual spacemacs-editing spacemacs-layouts spacemacs-completion spacemacs-defaults spacemacs-base spacemacs emacs-lisp c-c++ helm multiple-cursors treemacs))
  configuration-layer//load()
  (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configu..." "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading p...")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
  configuration-layer/load()
  (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
  eval-buffer(#<buffer  *load*> nil "/home/chz/.emacs.d/init.el" nil t)  ; Reading at buffer position 3159
  load-with-code-conversion("/home/chz/.emacs.d/init.el" "/home/chz/.emacs.d/init.el" t t)
  load("/home/chz/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x1417673be7823cda>) #f(compiled-function () #<bytecode -0x1f3c692ddc0ebef5>) t)
  command-line()
  normal-top-level()
```

And I found the reason:

https://github.com/syl20bnr/spacemacs/blob/9da9e3ff01d7071a894acc34e31d8c9a6dc983fe/layers/%2Blang/c-c%2B%2B/config.el#L75

Commenting this line out will solve this issue, but I'm not sure whether this will affect other config or whether this is caused by other variables such as emacs version, so I'll let the developer fix it. @smile13241324

There is also a similar comment on https://github.com/syl20bnr/spacemacs/commit/85d0aa2b18c07ef46c51e78fa266077c04f846e0 page which introducing this line.

FYI:
OS: Manjaro-Linux
Emacs version: GNU Emacs 28.0.50
spacemacs commit: 9da9e3ff01d7071a894acc34e31d8c9a6dc983fe

**Comment by stkain** [comment]
Hello c02y,
I'm facing the same problem.
There is already an issue open for this with a potential fix: #14631

**Comment by c02y** [comment]
Don't know why I failed to find it before I filed this issue, I remembered I searched this github repo and Google but couldn't find a similar issue like this, so I have to analyze myself, really spent some time to find `define-obsolete-variable-alias`.

I'll close it then since it is duplicaed.
